src/label/labelLayoutHelper.ts

```diff
@@ -321,6 +321,13 @@ export function hideOverlap(labelList: LabelLayoutInfo[]) {
         globalRect.height -= 0.1;
         globalRect.x += 0.05;
         globalRect.y += 0.05;
+        // if a labels bounding rect has no width or height (i.e. no text)
+        // it should not be shown
+        if (globalRect.width <= 0 && globalRect.height <= 0) {
```

Comment:
`globalRect` is the unioned bounding rect of the current label and previous labels. So should it be wiser to check `labelItem.rect.width === 0 || labelItem.rect.height === 0` instead, before `globalRect.copy(labelItem.rect)`?